include/enum_quda.h

```diff
@@ -526,6 +526,8 @@ extern "C" {
     QUDA_EXTLIB_INVALID = QUDA_INVALID_ENUM
   } QudaExtLibType;
 
+  enum class dslash_test_type { Dslash = 0, MatPC, Mat, MatPCDagMatPC, MatDagMat, M5, M5inv, Dslash4pre };
```

Comment:
This is C++ code - enum_quda.h must be a pure C header since it is part of the interface

Reply:
Good point. Reason it is there (apart from me forgetting about not putting c++ in the interface here) is that there are test only functions in the interface like `void dslashQuda_mdwf(void *h_out, void *h_in, QudaInvertParam *inv_param, QudaParity parity, dslash_test_type test_type)`.
Probably best to move them into a helper test thing.